Enable rt721 support for PTL-RVP

naveen-manohar opened the pull request:
Patch adds match table for rt721 multiple function codec on link3.

BUG=b:368495490
Test=WCL-SDV PO audio tests, verified on PTL-RVP board with RT721 Eval

bardliao commented on sound/soc/sdw_utils/soc_sdw_rt721_sdca.c:
```diff
@@ -0,0 +1,40 @@
+	{ "Speaker", NULL, "rt721 SPK" },
+};
+
+int asoc_sdw_rt721_spk_rtd_init(struct snd_soc_pcm_runtime *rtd, struct snd_soc_dai *dai)
```
Can we merge those Realtek multi-function codecs spk_rtd_init? Like what we did in soc_sdw_rt_amp.c. Currently, we have soc_sdw_rt712_sdca.c, soc_sdw_rt722_sdca.c, and soc_sdw_rt721_sdca.c, and I am sure there will be more coming.

naveen-manohar replied:
Agree, shall respin patch..

naveen-manohar committed:
ASoC: intel/sdw_utils: refactor RT multifunction sdca speaker codecs

Merge spk_rtd_init for multifunction sdca codecs:rt712/rt722

Signed-off-by: Naveen Manohar <naveen.m@intel.com>

naveen-manohar committed:
ASoC: Intel: soc-acpi-intel-ptl-match: Add rt721 support

Patch adds driver data & match table for rt721
multi-function codec on PTL-RVP at sdw link3.

Signed-off-by: Naveen Manohar <naveen.m@intel.com>

bardliao commented on sound/soc/sdw_utils/soc_sdw_rt_mf_sdca.c:
```diff
@@ -0,0 +1,74 @@
+#include <sound/soc-dapm.h>
+#include <sound/soc_sdw_utils.h>
+
+#define CODEC_NAME_SIZE	7
```
Should it be 6? The codec name will be "rt712" , "rt722", etc.

bardliao commented on sound/soc/sdw_utils/soc_sdw_rt_mf_sdca.c:
```diff
@@ -0,0 +1,74 @@
+	int ret;
+	int i;
+
+	rt_mf_sdca_map = get_codec_name_and_route(dai, codec_name);
```
`get_codec_name_and_route` could return NULL. So, we need to test `if (!rt_mf_sdca_map)`.

bardliao commented on sound/soc/sdw_utils/soc_sdw_rt_mf_sdca.c:
```diff
@@ -0,0 +1,74 @@
+			ret = snd_soc_dapm_add_routes(&card->dapm, rt_mf_sdca_map, 2);
+		else if (strstr(codec_dai->component->name_prefix, "-2"))
+			ret = snd_soc_dapm_add_routes(&card->dapm, rt_mf_sdca_map + 2, 2);
+	}
```
I don't think there will be "-1" or "-2" for multi function speakers. Just add `rt_mf_sdca_map` with `ARRAY_SIZE(rt_mf_sdca_map)`

naveen-manohar replied:
respined with a different logic, verified RT722 & RT721 on PTL
please check this

bardliao commented on sound/soc/sdw_utils/soc_sdw_rt_mf_sdca.c:
```diff
@@ -0,0 +1,87 @@
+		return ret;
+	}
+
+	return 0;
```
You could
```c
	if (ret)
		dev_err(rtd->dev, "failed to add rt sdca spk map: %d\n", ret);

	return ret;
```

naveen-manohar replied:
sure, done..

ujfalusi commented on sound/soc/intel/common/soc-acpi-intel-ptl-match.c:
```diff
@@ -69,6 +69,24 @@ static const struct snd_soc_acpi_adr_device rt711_sdca_0_adr[] = {
+		.num_adr = ARRAY_SIZE(rt721_3_single_adr),
+		.adr_d = rt721_3_single_adr,
+	},
+	{}
```
nitpick: Would be nice to be consistent within a file and add a comma after the sentinel

naveen-manohar replied:
thank you,
done
